### ISSUE TITLE
"All object keys must match" error on Insert

### ISSUE THREAD

**Issue by billypap1** [supabase-community/postgrest-swift#26]
Hello, I have integrated Supabase-Swift to my project.
When I try to insert a row, I receive this error - > "All object keys must match"  
Even if I include in my Codable Item values like id which is auto generated in db.

**Comment by srector13** [comment]
> 

I get the same thing in my setup. Still trying to get around this issue

**Comment by billypap1** [comment]
@srector13 Please inform me if you find any workaround

**Comment by billypap1** [comment]
@srector13  I found a workaround: 

Let's suppose you have this code: 

struct Todo: Codable {
    var text: String
    var integerval: String (make int to String)
    var boolval: String (make bool to string)
}


let todo = Todo(text: "Example todo!", integerval: String(123) , boolval: "false")

let jsonData: Data = try JSONEncoder().encode(todo)
let jsonDict = try JSONSerialization.jsonObject(with: jsonData, options: .allowFragments)

client.database.from("todos")    
    .insert(values: jsonDict as! [String:String])
    .execute { results in
        // Handle response
    }

All you have to do is to convert the Codable's Item Structure to String for all inputs even if they are Boolean, Int etc.

**Comment by grdsdev** [comment]
Hey guys, thanks for reporting this, that workaround shouldn't be needed, let me take a look and go back to you once I have a solution.

Thanks!

**Comment by grdsdev** [comment]
@billypap1 and @srector13 can you point the `supabase-swift` library to the `master` branch and check if that solves your issues?

**Comment by billypap1** [comment]
@grsouza I updated the package and tried with the 0.0.2 version, again the same issue -> "All object keys must match" .
Only If I set my values to String works.

PS. I use this library for db operations, not supabase-swift

**Comment by grdsdev** [comment]
@billypap1 I've setup an integration tests along with a local supabase instance in the `master` branch of this repo, take a look at https://github.com/supabase-community/postgrest-swift/blob/master/Tests/PostgRESTIntegrationTests/IntegrationTests.swift and you should see that it's using a `Codable` object with `non String` attributes and its working.

Can you provide a sample project that contains the issue you're having? Or point me at your project if it's public. That way I can clone it and debug it on my machine.

**Comment by steve-chavez** [comment]
FYI, we worked around this issue on postgrest-js with https://github.com/supabase/postgrest-js/issues/175.

**Comment by SergioB-dev** [comment]
FWIW I just put this package `PostgREST 0.0.2` into my iOS project and I get the same error.
Of the above proposed solutions the only one that is working is what  @srector13 suggested as a workaround.

 I would be happy to provide a sample project but at this point its literally the same thing as @srector13's suggestion

**Comment by richthornton** [comment]
@grsouza I was getting the same issue, and just tried the latest `master` (I'm new to Swift so had to look up how to install a package from a branch) and it worked great for me!

I followed the unit tests you linked above, which were super helpful as well (the new `async` `await` api is very nice 😍). Thanks so much!

**Comment by SergioB-dev** [comment]
@richthornton can you explain? Is 0.0.2 not the latest?

**Comment by richthornton** [comment]
@SergioB-dev apologies, I presumed that the `master` branch had some additional changes than 0.0.2, but looking at the commits it seems like it's mostly additional test cases. I just tried 0.0.2, and it still works (for me at least) so its not difference between master and 0.0.2

The other change I made at the same time was changing the code to follow the format that was in the test cases. Using @billypap1's example above, the code would be:
```
struct Todo: Codable {
    var text: String
    var integer: Int
    var bool: Bool
}

let todo = Todo(text: "Example todo!", integer: 123, bool: false)

client.from("todos")
    .insert(values: todo)
    .execute { results in
        print("Response: \(results)")
    }
```
The difference is removing the json encoding and serialisation lines. When I try the two versions locally, this change seems to fix the issue for me. Hopefully it solves it for you as well 🤞. If not if you post another minimal reproduction I can try helping narrow down the differences between our setups.
